### PR TITLE
Take project Name from filename if missing; warn if the Name and the filename do not match

### PR DESCRIPTION
**Pull request by matyasselmeci**
SOFTWARE-4755

### CODE REVIEW

**Comment by matyasselmeci** [comment]
Also, these are the warnings it found (edited for brevity):
```
projects/UCAnschutz_Langner.yaml: 'Name' 'UCAnshutz_Langner' does not match filename
projects/GSU_Maimon.yaml: 'Name' 'GSU_Wang' does not match filename
projects/TNTTech_ITS.yaml: 'Name' 'TNTech_ITS' does not match filename
projects/PRP_Defanti.yaml: 'Name' 'PRP' does not match filename
projects/NMSU_staff.yaml: 'Name' 'NMSU_Staff' does not match filename
```

**Comment by brianhlin** [comment]
@lmichael107 @ChristinaLK @rynge @eharstad any idea which ones are the right project names from @matyasselmeci's comment above?

File: src/webapp/project_reader.py [hunk 1]
```diff
@@ -65,6 +65,14 @@ def get_one_project(file: str, campus_grid_ids: Dict, vos_data: VOsData) -> Dict
             data['ResourceAllocations'] = {"ResourceAllocation": resource_allocations}
         if 'ID' not in data:
             del project['ID']
+
+        name_from_filename = os.path.basename(file)[:-5]  # strip '.yaml'
```
Comment (brianhlin) [inline src/webapp/project_reader.py:69]: What about `.yml`?

**Review by brianhlin — APPROVED** [review]
One comment but otherwise LGTM

File: src/webapp/project_reader.py (hunk 1 above)
Comment (matyasselmeci) [reply src/webapp/project_reader.py:69]: We don't read .yml files; this goes for the VO and resource readers as well.

**Pull request merged**